vermouth/processors/stash_attributes.py

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+# Copyright 2025 University of Groningen
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Provides a processor that stores attributes 
+"""
+
+from .processor import Processor
+from ..log_helpers import StyleAdapter, get_logger
+
+LOGGER = StyleAdapter(get_logger(__name__))
+
+
+def stash_attributes(molecule, attributes):
+    """
+    For each node in molecule, add the attributes to a stash dictionary
+
+    Parameters
+    ----------
+    molecule: :class:`~vermouth.molecule.Molecule`
+        The molecule to transform.
+    attributes: tuple[str]
+        Attributes to store in the nodes that may otherwise be modified
+    """
+    for attr in attributes:
+        for node in molecule.nodes:
```

Comment:
```suggestion
    for node in molecule.nodes:
        for attr in attributes:
```
For me this makes the intent more clear: for each node, store these attributes